src/cephadm/cephadm.py

```diff
@@ -1488,30 +1487,135 @@ def run(self) -> None:
+            logger.info(f'Received node_proxy_oob_cmd command: {node_proxy_cmd}')
+            if node_proxy_cmd['action'] == 'get_led':
+                if node_proxy_cmd['type'] == 'chassis':
+                    self.node_proxy_oob_cmd_result = self.agent.node_proxy_mgr.node_proxy.system.get_chassis_led()
+            if node_proxy_cmd['action'] == 'set_led':
+                if node_proxy_cmd['type'] == 'chassis':
+                    _data: Dict[str, Any] = json.loads(node_proxy_cmd['data'])
+                    _result: int = self.agent.node_proxy_mgr.node_proxy.system.set_chassis_led(_data)
+                    self.node_proxy_oob_cmd_result = {'http_code': _result}
+        else:
+            raise RuntimeError('No valid data received.')
+
+
+class NodeProxyManager(Thread):
+    def __init__(self, agent: 'CephadmAgent', event: Event):
+        super().__init__()
+        self.agent = agent
+        self.event = event
+        self.stop = False
+
+    def run(self) -> None:
+        self.event.wait()
+        self.ssl_ctx = self.agent.ssl_ctx
+        self.init()
+        self.loop()
+
+    def init(self) -> None:
+        node_proxy_meta = {
+            'cephx': {
+                'name': self.agent.host,
+                'secret': self.agent.keyring
+            }
+        }
+        status, result = http_query(addr=self.agent.target_ip,
+                                    port=self.agent.target_port,
+                                    data=json.dumps(node_proxy_meta).encode('ascii'),
+                                    endpoint='/node-proxy/oob',
+                                    ssl_ctx=self.ssl_ctx)
+        if status != 200:
+            msg = f"Couldn't load oob tool details: {status}, {result}"
+            logger.debug(msg)
+            raise NodeProxyFetchOobError(msg)
+        result_json = json.loads(result)
+        kwargs = {
+            'host': result_json['result']['addr'],
+            'username': result_json['result']['username'],
+            'password': result_json['result']['password'],
+            'cephx': node_proxy_meta['cephx'],
+            'mgr_target_ip': self.agent.target_ip,
+            'mgr_target_port': self.agent.target_port
+        }
+        if result_json['result'].get('port'):
+            kwargs['port'] = result_json['result']['port']
+
+        self.node_proxy: NodeProxy = NodeProxy(**kwargs)
+        self.node_proxy.start()
+
+    def loop(self) -> None:
+        while not self.stop:
+            try:
+                status = self.node_proxy.check_status()
+                label = 'Ok' if status else 'Critical'
+                logger.debug(f'node-proxy status: {label}')
+            except NodeProxyFetchOobError:
+                logger.info('No out of band tool details could be loaded. '
+                            'Aborting node-proxy initialization. '
+                            'Will retry in 120s.')
+                time.sleep(120)
+                self.init()
```

Comment:
It seems this `init` can also raise a `NodeProxyFetchOobError`. Do we need error handling for that as well? Or is this just meant to run at startup and if init fails then it goes down?

Reply:
yeh, i think something is wrong with this, it needs to be reworked a bit.

Reply:
reworked